pkg/handlers/adminapi/webhook_subscriptions.go

```
@@ -91,3 +93,31 @@ func (h GetWebhookSubscriptionHandler) Handle(params webhooksubscriptionop.GetWe
 	payload := payloadForWebhookSubscriptionModel(webhookSubscription)
 	return webhooksubscriptionop.NewGetWebhookSubscriptionOK().WithPayload(payload)
 }
+
+// UpdateWebhookSubscriptionHandler returns an updated webhook subscription via PUT
+type UpdateWebhookSubscriptionHandler struct {
+	handlers.HandlerContext
+	services.WebhookSubscriptionUpdater
+	services.NewQueryFilter
+}
+
+// Handle updates a webhook subscription content
+func (h UpdateWebhookSubscriptionHandler) Handle(params webhooksubscriptionop.UpdateWebhookSubscriptionParams) middleware.Responder {
+	logger := h.LoggerFromRequest(params.HTTPRequest)
+	payload := params.WebhookSubscription
+
+	// Payload to model converter
+	webhookSubscription := payloads.WebhookSubscriptionModel(payload)
+	logger.Info("got here")
+	fmt.Println(webhookSubscription)
+
+	updatedWebhookSubscription, err := h.WebhookSubscriptionUpdater.UpdateWebhookSubscription(webhookSubscription, "eTag")
+	// Check that the uuid provided is valid
+	if err != nil {
+		logger.Error(fmt.Sprintf("Error updating webhookSubscription %s", params.WebhookSubscriptionID.String()), zap.Error(err))
```

If the update failed, probably you want to return some sort of error response? This way it eventually returns an ok, even if the update failed.

---

After this err check if there was something wrong, would there be another failure that hasn't been checked? Or does this error response need to be more specific?

---

No I meant this "logger.Error" is just printing something in the logs, you need to return an actual error to the client. See other admin api handlers to see how they return errors (since it's probably a bit different than primeAPI).
```
return webhooksubscriptionop.NewUpdateWebhookSubscriptionInternalServerError...
```